src/viam/sdk/common/utils.hpp

```diff
@@ -44,5 +44,8 @@ class ResourceNameEqual {
 std::vector<unsigned char> string_to_bytes(std::string const& s);
 std::string bytes_to_string(std::vector<unsigned char> const& b);
 
+std::chrono::duration<double> duration_from_proto(const google::protobuf::Duration& proto);
+google::protobuf::Duration duration_to_proto(const std::chrono::duration<double>& duration);
```

Comment:
These feel very awkward here. I cannot call them `from_proto` in global namespace as otherwise there are discovery problems

If people think this is a problem, I would propose either:
1) a shared `common/duration.hpp` in a new `common` namespace (move Vector3 into this new namespace as well)
2) a `viam::cpp::duration` class that wraps `std::chrono::duration<double>` (I think this would be confusing to the user)

Reply:
I'm not quite sure what you mean by "discovery problems", can you elaborate?

Reply:
If I remove the `duration_` prefix in utils.hpp and utils.cpp I get the following error:
```
v/nix/store/dkw46jgi8i0bq64cag95v4ywz6g9bnga-gcc-wrapper-11.3.0/bin/g++ -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -DBOOST_CHRONO_DYN_LINK -DBOOST_CHRONO_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_FILESYSTEM_NO_LIB -DBOOST_LOG_DYN_LINK -DBOOST_LOG_NO_LIB -DBOOST_REGEX_DYN_LINK -DBOOST_REGEX_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -Dviamsdk_EXPORTS -I/home/zack/work/viam-cpp-sdk/build/src/viam/api/gen/viam/api/../.. -I/home/zack/work/viam-cpp-sdk/build/src/viam/api/gen/viam/api -I/home/zack/work/viam-cpp-sdk/src/viam/sdk/../.. -g -fPIC -Wall -Werror -std=c++14 -MD -MT src/viam/sdk/CMakeFiles/viamsdk.dir/components/board/client.cpp.o -MF src/viam/sdk/CMakeFiles/viamsdk.dir/components/board/client.cpp.o.d -o src/viam/sdk/CMakeFiles/viamsdk.dir/components/board/client.cpp.o -c /home/zack/work/viam-cpp-sdk/src/viam/sdk/components/board/client.cpp
/home/zack/work/viam-cpp-sdk/src/viam/sdk/components/board/client.cpp: In member function ‘virtual void viam::sdk::BoardClient::set_power_mode(viam::sdk::Board::power_mode, const std::chrono::duration<double>&)’:
/home/zack/work/viam-cpp-sdk/src/viam/sdk/components/board/client.cpp:194:43: error: no matching function for call to ‘viam::sdk::BoardClient::to_proto(const std::chrono::duration<double>&)’
  194 |     *request.mutable_duration() = to_proto(duration);
      |                                   ~~~~~~~~^~~~~~~~~~
```

Reply:
The board client tries to find a locally (i.e. Board) overloaded `to_proto` rather than the global utils.hpp `to_proto`

Reply:
I'll bet you can make them visible with a `using` statement:

```
void viam::sdk::BoardClient::set_power_mode(...) {
    using ::viam::to_proto;
    
}
```

Reply:
Essentially, you are running into the fact that `from_proto` and `to_proto` are overloaded at different scopes so the resolution gets a little weird.

I think go with explicitly qualifying it for now, and keep the name consistent. I overall have some interest in how the proto conversion stuff evolves but I don't think now is the time to pursue it.